src/zenml/zen_server/routers/users_endpoints.py

```diff
@@ -238,7 +247,11 @@ def update_user(
         """
         user = zen_store().get_user(user_name_or_id)
         if user.id != auth_context.user.id:
-            verify_permission_for_model(user, action=Action.UPDATE)
+            verify_permission_for_model(
+                user,
+                action=Action.UPDATE,
+                admin_status=auth_context.user.is_admin,
+            )
```

Comment:
For non-admin users, we also have to prevent users from updating security-sensitive fields in their user account:
* `is_admin` !!! <- this I already checked and any user is currently able to elevate their account to admin
* `active` -> to prevent regular users from deactivating their own account
* `activation_token` -> just in case

You should just clear these fields in the incoming update, in case the user is not admin.

Maybe we can even generalize some rules and say no account should be allowed to deactivate itself or to set/reset the activation token. Up to you.

Please remember to update the `update_myself` endpoint too. This is critical.

Reply:
Come to think of it, we should definitely have integration tests for these :smile:

Reply:
Heavily reworked this part.